File: solutions/leetcode_201.py
class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        shift = 0
        while 0 < left < right:
            left >>= 1
            right >>= 1
            shift += 1
        return left << shift


if __name__ == '__main__':
    given_left = 1
    given_right = 2147483647
    expected = 0

    solution = Solution()
    result = solution.rangeBitwiseAnd(given_left, given_right)
    assert result == expected

File: solutions/leetcode_201.swift
class Solution {
    func rangeBitwiseAnd(_ left: Int, _ right: Int) -> Int {
        var left = left
        var right = right
        var shift = 0
        while left > 0 && left < right {
            left >>= 1
            right >>= 1
            shift += 1
        }
        return left << shift
    }
}


let given_left = 1
let given_right = 2147483647
let expected = 0

let solution = Solution()
let result = solution.rangeBitwiseAnd(given_left, given_right)
if result != expected {
    print("\(result) not equals \(expected)")
}
